src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -173,4 +184,40 @@ private static boolean match(String one, String two) {
     two = two.toLowerCase().trim();
     return one.contains(two) || two.contains(one);
   }
-}
+
+  /**
+   * Returns product name from {@link CpeMatch#getCpe23Uri()}.
+   */
+  private String productFrom23Uri(String cpe23Uri) {
+    return cpe23Uri.split(":")[4];
+  }
+
+  /**
+   * Returns product name from {@link CpeMatch#getCpe22Uri()}.
+   */
+  private String productFrom22Uri(String cpe22Uri) {
+    return cpe22Uri.split(":")[3];
+  }
```

Comment:
Maybe it ArrayIndexOutOfBoundsException won't happen here, but please make sure that those elements in the array exist before accessing them.

Reply:
You are right.